### PR TITLE
Fix custom meter error with zone mechanical ventilation and airloop variables

### LINKED ISSUE #7655: Meter:Custom Variable Error
Issue overview
--------------
I tried to add Meter:Custom objects as:

```
Meter:Custom,
  Building_Cooling_Coil_Total_Cooling_Energy, !- Name
  Generic,                                !- Fuel Type
  *,                                      !- Key Name 1
  Cooling Coil Total Cooling Energy;      !- Output Variable or Meter Name 1

Output:Meter,
  Building_Cooling_Coil_Total_Cooling_Energy, !- Key Name
  timestep;                               !- Reporting Frequency

Meter:Custom,
  Building_Zone_Mechanical_Ventilation_Cooling_Load_Decrease_Energy, !- Name
  Generic,                                !- Fuel Type
  *,                                      !- Key Name 1
  Zone Mechanical Ventilation Cooling Load Decrease Energy; !- Output Variable or Meter Name 1

Output:Meter,
  Building_Zone_Mechanical_Ventilation_Cooling_Load_Decrease_Energy, !- Key Name
  timestep;                               !- Reporting Frequency
```

The first 'Cooling Coil Total Cooling Energy' works well when the second gives error as:

```
 ** Warning ** Meter:Custom="BUILDING_ZONE_MECHANICAL_VENTILATION_COOLING_LOAD_DECREASE_ENERGY", invalid Output Variable or Meter Name="ZONE MECHANICAL VENTILATION COOLING LOAD DECREASE ENERGY".
   **   ~~~   ** ...will not be shown with the Meter results.
   ** Warning ** Meter:Custom="BUILDING_ZONE_MECHANICAL_VENTILATION_COOLING_LOAD_DECREASE_ENERGY", no items assigned 
   **   ~~~   ** ...will not be shown with the Meter results. This may be caused by a Meter:Custom be assigned to another Meter:Custom.
   ** Warning ** Output:Meter: invalid Key Name="BUILDING_ZONE_MECHANICAL_VENTILATION_COOLING_LOAD_DECREASE_ENERGY" - not found.
```
### Details
 - Platform (Operating system, version): Mac OS 10.14.6
 - Version of EnergyPlus (if using an intermediate build, include SHA): 9.2
 - Unmethours link or helpdesk ticket number: similar to this issue  [(unmethour_link)](https://unmethours.com/question/38038/using-metercustom-to-aggregate-output-variable-data/)
### Checklist
Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
 - [x] Defect file added [(url_idf_file)](https://drive.google.com/file/d/10xOyDlc4ro_dv6zCSG2cG2OBsIoqVC-Q/view?usp=sharing)

### PR DESCRIPTION
**Pull request by LipingWang**
Pull request overview
---------------------
 - Fixes #7655 
 - Fixed the issue with custom meter.

### Pull Request Author
Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
 - [x] Title of PR should be user-synopsis style (clearly understandable in a standalone changelog context)
 - [x] Label the PR with at least one of: Defect, Refactoring, NewFeature, Performance, and/or DoNoPublish
 - [ ] Pull requests that impact EnergyPlus code must also include unit tests to cover enhancement or defect repair
 - [ ] Author should provide a "walkthrough" of relevant code changes using a GitHub code review comment process
 - [ ] If any diffs are expected, author must demonstrate they are justified using plots and descriptions
 - [ ] If changes fix a defect, the fix should be demonstrated in plots and descriptions
 - [ ] If any defect files are updated to a more recent version, upload new versions here or on DevSupport
 - [ ] If IDD requires transition, transition source, rules, ExpandObjects, and IDFs must be updated, and add IDDChange label
 - [ ] If structural output changes, add to output rules file and add OutputChange label
 - [ ] If adding/removing any LaTeX docs or figures, update that document's CMakeLists file dependencies

### Reviewer
This will not be exhaustively relevant to every PR.
 - [ ] Perform a Code Review on GitHub
 - [ ] If branch is behind develop, merge develop and build locally to check for side effects of the merge
 - [ ] If defect, verify by running develop branch and reproducing defect, then running PR and reproducing fix
 - [ ] If feature, test running new feature, try creative ways to break it
 - [ ] CI status: all green or justified
 - [ ] Check that performance is not impacted (CI Linux results include performance check)
 - [ ] Run Unit Test(s) locally
 - [ ] Check any new function arguments for performance impacts
 - [ ] Verify IDF naming conventions and styles, memos and notes and defaults
 - [ ] If new idf included, locally check the err file and other outputs

### CODE REVIEW

**Comment by Myoldmopar** [comment]
New warnings showing up in the error files with this change.  It looks like there are side effects to making this change.

**Comment by mjwitte** [comment]
Timing is everything during initial setup. Will rethink this - we may need to split up what `CreateEnergyReportStructure` does so that the output variables are present when `UpdateMeterReporting` is called.

File: src/EnergyPlus/SimulationManager.cc [hunk 1]
```diff
@@ -281,6 +281,7 @@ namespace SimulationManager {
         using SizingManager::ManageSizing;
         using SystemReports::CreateEnergyReportStructure;
         using SystemReports::ReportAirLoopConnections;
+        using SystemReports::AllocateAndSetUpVentReports;
```
Comment (mjwitte) [inline src/EnergyPlus/SimulationManager.cc]: @LipingWang Minor detail. We're trying to move away from `using` statements and be explicit when actually using the function or variable. So, when you add the unit test, please delete this line.

File: src/EnergyPlus/SimulationManager.cc [hunk 2]
```diff
@@ -410,12 +411,12 @@ namespace SimulationManager {
             SetupNodeVarsForReporting();
             MetersHaveBeenInitialized = true;
             SetupPollutionMeterReporting();
+            AllocateAndSetUpVentReports();
```
Comment (mjwitte) [inline src/EnergyPlus/SimulationManager.cc]: And here change to `SystemReports::AllocateAndSetUpVentReports();`

File: src/EnergyPlus/SimulationManager.cc [hunk 3]
```diff
@@ -428,7 +429,7 @@ namespace SimulationManager {
             if (ErrFound) TerminalError = true;
             CheckControllerLists(ErrFound);
             if (ErrFound) TerminalError = true;
-
+
```
Comment (mjwitte) [inline src/EnergyPlus/SimulationManager.cc]: Very minor - remove trailing spaces.

**Review by mjwitte — COMMENTED** [review]
This looks good. Now it needs a new unit test. `OutputProcessor.unit.cc:: TEST_F(EnergyPlusFixture, OutputProcessor_DuplicateMeterCustom)` would be a good starting point.

**Comment by mitchute** [comment]
@mjwitte it looks like you've already been through this. I did build and run the unit tests locally, and it looks good. I also went ahead and ran clang-format on ```OutputProcessors.unit.cc``` but I probably won't do that again since it muddles up the changelog. FFR the unit test ```OutputProcessor_MeterCustomSystemEnergy``` was added during this PR.

I'll let CI run again, but this should be ready to merge.

**Comment by mjwitte** [comment]
@mitchute I helped @LipingWang develop this fix, so someone else (like you) should be doing the full review on this.

**Comment by mitchute** [comment]
Looks good. Merging.

**Comment by mitchute** [comment]
@LipingWang @mjwitte this looks good. I just ran the defect IDF file [(here)](https://github.com/NREL/EnergyPlusDevSupport/blob/54a4c4d817dfed8cffec1be0ff57ba6908baf549/DefectFiles/7000s/7655/5ZoneFanCoilDOASCool_test_meter_output_out.idf) and verified that the extra error file warning messages are now gone. Merging.

**Pull request merged**